File: 3340/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(isBalanced("1234"))
	fmt.Println(isBalanced("2222"))
}

func isBalanced(num string) bool {
	o := 0
	j := 0
	for i, v := range num {
		if i % 2 == 0 {
			o += int(v - '0')
		} else {
			j += int(v - '0')
		}
	}
	return o == j
}

File: 3340/main.rs
struct Solution{}

impl Solution {
    pub fn is_balanced(num: String) -> bool {
        let mut df = 0;
        let mut s = 1;
        for c in num.chars() {
            let d = c.to_digit(10).unwrap() as i32;
            df += d * s;
            s = -s;
        }
        df == 0
    }
}

fn main() {
    println!("{:#?}", Solution::is_balanced("1234".to_string()));
}